fileblock/src/simplefs.rs

```diff
@@ -0,0 +1,25 @@
+// slightly arbitrary size of our "disk": 4GIB, broken into 4KiB blocks
+// 4 * 2.pow(30) / 4096 == 2Kib required to store bitmap
+const BITMAP_LEN: usize = 1048576_usize;
+
+/// disk block free list 
+/// persistent tracking of what blocks have been allocated.
+/// option 1: Bitmap! naive style
+struct  DiskBlockFreelist {
```

Comment:
an option 1.5 would be a "packed" bitmap

which would be better on space at least. 

[here's an unsafe rust implmentation from a long time ago that is a good reference](https://github.com/emberian/rust-bitmap/blob/master/src/lib.rs)

Reply:
Bitmask is probably the way to go. There are probably upper bound based on physical size (ext2 only supported up to 4TB hard disks), that's a good guide for the design here.

Our block could look like three discrete and continuous superblocks:

| block accounting data structure | inode fs (metadata) | storage block (n number of blocks) |

We can track where our accounting structure is by the ordering, maybe fixing the side of the first two blocks is necessary... I'm not sure there.

Reply:
If we do use a fixed size array, how do we know how many blocks are actually available to us when loading this from storage?

Reply:
> f we do use a fixed size array, how do we know how many blocks are actually available to us when loading this from storage?

I think you'd have to scan the entire array for the empty bits when you boot up, at least in the naive implementation